sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1129 @@
+    """A metadata is requested either with a string alias or this enum.
+
+    .. versionadded:: 1.1
+    """
+
+    # Metadata is not requested. It will not be routed to the object having the
+    # request value as UNREQUESTED.
+    UNREQUESTED = False
+    # Metadata is requested, and will be routed to the requesting object. There
+    # will be no error if the metadata is not provided.
+    REQUESTED = True
+    # Default metadata request configuration. It should not be passed, and if
+    # present, an error is raised for the user to explicitly set the request
+    # value.
+    ERROR_IF_PASSED = None
+    # this is used in `__metadata_request__*` attributes to indicate that a
+    # metadata is not present even though it may be present in the
+    # corresponding method's signature.
+    UNUSED = "$UNUSED$"
+    # this is used whenever a default value is changed, and therefore the user
+    # should explicitly set the value, otherwise a warning is shown. An example
+    # is when a meta-estimator is only a router, but then becomes also a
+    # consumer.
+    WARN = "$WARN$"
+
+
+# this is the default used in `set_{method}_request` methods to indicate no change
+# requested by the user.
+UNCHANGED = "$UNCHANGED$"
+
+# Only the following methods are supported in the routing mechanism. Adding new
+# methods at the moment involves monkeypatching this list.
+METHODS = [
+    "fit",
+    "partial_fit",
+    "predict",
+    "score",
+    "split",
+    "transform",
+    "inverse_transform",
+]
+
+
+# These strings are used to dynamically generate the docstrings for
+# set_{method}_request methods.
+REQUESTER_DOC = """        Request metadata passed to the ``{method}`` method.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.1
+
+        Parameters
+        ----------
+"""
+REQUESTER_DOC_PARAM = """        {metadata} : RequestType, str, True, False, or None, \
+                    default=UNCHANGED
+            Whether {metadata} should be passed to {method} by meta-estimators or
+            not, and if yes, should it have an alias.
+
+            - True or RequestType.REQUESTED: {metadata} is requested, and passed to \
+{method} if provided.
+
+            - False or RequestType.UNREQUESTED: {metadata} is not requested and the \
+meta-estimator will not pass it to {method}.
+
+            - None or RequestType.ERROR_IF_PASSED: {metadata} is not requested, and \
+the meta-estimator will raise an error if the user provides {metadata}
+
+            - str: {metadata} should be passed to the meta-estimator with this given \
+alias instead of the original name.
+
+            The default (UNCHANGED) retains the existing request. This allows
+            you to change the request for some parameters and not others.
+
+"""
+REQUESTER_DOC_RETURN = """        Returns
+        -------
+        self : estimator
+            Returns the object itself.
```

Comment:
```suggestion
            The updated estimator.
```

Reply:
If we use this on a splitter, it doesn't return an estimator.

Reply:
The line before says "self : estimator", or else I wouldn't have suggested so :)